in_toto/util.py

```diff
@@ -5,58 +5,42 @@
 import in_toto.gpg.formats
 import securesystemslib.formats
 import securesystemslib.hash
+import securesystemslib.interface
 import securesystemslib.keys
 import securesystemslib.exceptions
 
+from securesystemslib.interface import (import_ed25519_privatekey_from_file,
+    import_ed25519_publickey_from_file)
+
 DEFAULT_RSA_KEY_BITS = 3072
 
-def generate_and_write_rsa_keypair(filepath, bits=DEFAULT_RSA_KEY_BITS,
-    password=None):
-  """
-  <Purpose>
-    Generate an RSA key keypair and store public and private portion each
-    to a file in PEM format.
-    If a password is specified the private key is encrypted using that password
-    as pass phrase.
-    The private key is stored to <'filepath'> and the public key to
-    <'filepath'>.pub
+KEY_TYPE_RSA = 'rsa'
+KEY_TYPE_ED25519 = 'ed25519'
 
-  <Arguments>
-    filepath:
-      The public and private key files are saved to <filepath>.pub, <filepath>
-      respectively
+SUPPORTED_KEY_TYPES = [KEY_TYPE_ED25519, KEY_TYPE_RSA]
 
-    password: (optional)
-      If specified the password is used to encrypt the private key
-
-  <Exceptions>
-    securesystemslib.exceptions.FormatError, if the arguments are
-    improperly formatted
 
-  <Side Effects>
-    Writes key files to '<filepath>' and '<filepath>.pub'
-
-  <Returns>
-    None.
+def generate_and_write_rsa_keypair(filepath, bits=DEFAULT_RSA_KEY_BITS,
+    password=""):
   """
-  securesystemslib.formats.PATH_SCHEMA.check_match(filepath)
-
-  rsa_key = securesystemslib.keys.generate_rsa_key(bits)
-
-  public = rsa_key["keyval"]["public"]
-  private = rsa_key["keyval"]["private"]
-
-  if password:
-    private_pem = securesystemslib.keys.create_rsa_encrypted_pem(
-        private, password)
-  else:
-    private_pem = private
+  Calls securesystemslib.interface.generate_and_write_rsa_keypair with
+  a default password of "" so as to prevent the library from prompting for
+  password. If prompting is needed, it's already handled here and then
+  forwarded.
+  """
+  securesystemslib.interface.generate_and_write_rsa_keypair(filepath, bits,
+      password)
 
-  with open(filepath + ".pub", "wb") as fo_public:
-    fo_public.write(public.encode("utf-8"))
 
-  with open(filepath, "wb") as fo_private:
-    fo_private.write(private_pem.encode("utf-8"))
+def generate_and_write_ed25519_keypair(filepath, password=""):
+  """
+  Calls securesystemslib.interface.generate_and_write_ed25519_keypair with
+  a default password of "" so as to prevent the library from prompting for
+  password. If prompting is needed, it's already handled here and then
+  forwarded.
+  """
```

Comment:
Thanks for including the docstring and for providing good info!  The only other advice I can give is to specify the exception that this function raise, expected return value, etc.

We also have a guideline for docstrings [here](
https://github.com/secure-systems-lab/code-style-guidelines#documentation-strings).